packages/wonder-blocks-core/components/provider-group.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import * as React from "react";
+
+type Provider<P> = {|
+    type: React.ComponentType<P>,
+    props?: P,
+|};
+
+type Props = {|
+    providers: Array<Provider<any>>,
+    children: (Array<any>) => React.Node,
+|};
+
+export default class ProviderGroup extends React.Component<Props> {
+    render() {
+        const {providers, children} = this.props;
+        const provided = [];
+        let i = 0;
+        const buildProviders = (...args) => {
+            if (i > 0) {
+                provided.push(...args);
+            }
+            if (i < providers.length) {
+                const providerEl = React.createElement(providers[i].type, {
+                    ...providers[i].props,
+                    children: buildProviders,
+                });
+                i++;
+                return providerEl;
+            } else {
+                return children(provided);
```

Comment:
If we have made it through all the providers, we return the result of the real children given the now gathered provided values (which are in the same order as the providers array passed to the props, ensuring an intuitive ordering for consumers).

Reply:
It would be good to include this and the previous two comments.

Reply:
Yeah, I know. Was just faster to do it here for now :)